### DISCUSSION TITLE
Can't Install @vidstack/react package

### DISCUSSION THREAD

**Discussion by aazmaine23** [vidstack/player#1502] [Q&A] [answered]
When I try to install vidstack react using 
`npm i @vidstack/react@next` 

It shows an error: 
`npm i @vidstack/react
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: out@0.1.0
npm error Found: react@19.0.0-rc-02c0e824-20241028
npm error node_modules/react
npm error   react@"19.0.0-rc-02c0e824-20241028" from the root project
npm error
npm error Could not resolve dependency:
npm error peer react@"^18.0.0" from @vidstack/react@0.6.15
npm error node_modules/@vidstack/react
npm error   @vidstack/react@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /home/.npm/_logs/2024-11-11T07_02_11_993Z-eresolve-report.txt

npm error A complete log of this run can be found in: /home/.npm/_logs/2024-11-11T07_02_11_993Z-debug-0.log` 

**How to fix this issue? I need to use vidstack for my own project. Please help me**

**Comment by Anatolii-Ovcharuk** [comment, marked as answer]
Error you’re seeing occurs because Vidstack’s @vidstack/react package requires React 18, but you currently have React 19.0.0-rc installed. Here are some ways to resolve the dependency conflict:

Option 1: Downgrade to React 18
Since @vidstack/react depends on React 18, you could downgrade React to a compatible version:
npm install react@18 react-dom@18
Then, try installing Vidstack:
npm install @vidstack/react@next

Option 2: Use --legacy-peer-deps Flag
If you want to try installing @vidstack/react without downgrading React, you can bypass the dependency check with the --legacy-peer-deps flag:
npm install @vidstack/react@next --legacy-peer-deps
However, this might lead to unexpected behavior, as the package isn’t fully tested with React 19.
